src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128_1.cs

```diff
@@ -298,7 +298,7 @@ public override int GetHashCode()
         public Vector64<T> GetLower()
         {
             ThrowIfUnsupportedType();
-            Vector64<T>.ThrowIfUnsupportedType();
```

Comment:
These checks exist because it is possible for a type to be supported by one vector size but not another.

This is "somewhat" the case today in that `double`, `long`, and `ulong` are supported by the software but have no hardware fallback (and it has been requested a couple times we review this).

It could also be the case for future types.

Reply:
> This is "somewhat" the case today in that double, long, and ulong are supported by the software but have no hardware fallback

`ThrowIfUnsupportedType` only checks for numeric types that "just works
", and they are all identical today. 

> It could also be the case for future types.

If it happens in the future, we need another approach to handle (maybe software fallback).

Reply:
We should probably continue having the correct checks and should try to address the inlining issue separately.

Reply:
I know these checks may be useful in the future, but totally redundant today...

Reply:
We still have to consider ARM and we still haven't added support for `Half`.

The point of having the checks was to ensure correctness both now and in the future. So, I would very much be in favor of fixing the inlining issue, if possible, first.

Otherwise, we should merge the `ThrowIfUnsupportedType` methods into a single helper method with a comment for splitting it out in the future, to help ensure it isnt broken and that these are explicitly shared right now.

Reply:
> we should merge the ThrowIfUnsupportedType methods into a single helper

Yes, we should go this way. For the platform-agnostic methods, it makes no sense to support a base type on a vector size but not support on another vector size.